configure.ac

```diff
@@ -0,0 +1,179 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+
+m4_define([MLTON_VERSION], [`date +%Y%m%d.%H%M%S`])
```

Comment:
I'm not sure what to recommend here.  As written, this seems to set the `MLTON_VERSION` to the time at which `autoconf` created `configure`, but that doesn't really reflect the version of the sources.  In `mlton/Makefile`, we use:
```
$$(TZ=UTC git log -n1 --date=format-local:"%Y%m%d.%H%M%S" --pretty=format:"%cd-g%h$$([ "$$(git status --porcelain 2> /dev/null)" ] && echo '-dirty')" 2> /dev/null || echo '????????')
```
to create a version number based on the last commit.  But again, doing something like that would still set the version to the commit at the time `autoconf` was run and not reflect commits that happened since `autoconf` was run (which, admittedly, is only an issue for a MLton developer; the typical MLton user would checkout and immediately build, with no commits pulled/made between `autoconf` and `make all`).

I think that we should at least compute the version string from the `git` context.

Reply:
Let me define the version within the `configure` step then: the `AC_INIT` macro in `configure.ac` will export `PACKAGE_VERSION` that I'll then reuse within `mlton/Makefile`